netbench/cdk/src/main/java/com/aws/StateMachineStack.java

```diff
@@ -2,18 +2,95 @@
 // SPDX-License-Identifier: Apache-2.0
 package com.aws;
 
+import software.constructs.Construct;
 import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.services.stepfunctions.tasks.EcsRunTask;
 import software.amazon.awscdk.services.stepfunctions.StateMachine;
-import software.constructs.Construct;
+import software.amazon.awscdk.services.s3.Bucket;
+import software.amazon.awscdk.Duration;
+import software.amazon.awscdk.services.logs.RetentionDays;
+
+import software.amazon.awscdk.services.stepfunctions.tasks.LambdaInvoke;
+import software.amazon.awscdk.services.stepfunctions.tasks.LambdaInvocationType;
+import software.amazon.awscdk.services.stepfunctions.Wait;
+import software.amazon.awscdk.services.stepfunctions.WaitTime;
+import software.amazon.awscdk.services.stepfunctions.tasks.ContainerOverride;
+import software.amazon.awscdk.services.stepfunctions.tasks.TaskEnvironmentVariable;
+import software.amazon.awscdk.services.stepfunctions.JsonPath;
+import software.amazon.awscdk.services.stepfunctions.IntegrationPattern;
+import software.amazon.awscdk.services.stepfunctions.tasks.EcsEc2LaunchTarget;
+
+import software.amazon.awscdk.services.ecs.ContainerDefinition;
+import software.amazon.awscdk.services.ecs.ContainerDefinitionOptions;
+import software.amazon.awscdk.services.ecs.Ec2TaskDefinition;
+import software.amazon.awscdk.services.ecs.ContainerImage;
+import software.amazon.awscdk.services.ecs.AwsLogDriverProps;
+import software.amazon.awscdk.services.ecs.LogDriver;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
 
 public class StateMachineStack extends Stack {
 
     public StateMachineStack(final Construct parent, final String id, final StateMachineStackProps props) {
         super(parent, id, props);
 
+        Bucket bucket = props.getBucket();
+
+        EcsRunTask clientTask = props.getClientTask();
+
+        Wait waitFunction = Wait.Builder.create(this, "wait-step")
+            .time(WaitTime.duration(Duration.seconds(20)))
+            .build();
+
+        Ec2TaskDefinition reportGenerationTask = Ec2TaskDefinition.Builder
+            .create(this, "report-generation-task")
+            .build();
+
+        Map<String, String> reportGenerationEnv = new HashMap<>();
+        reportGenerationEnv.put("S3_BUCKET", bucket.getBucketName());
```

Comment:
From what I see, this stack depends on the vpc stack because it needs the s3 bucket name. If this is the case then we can actually remove this dependency all together by declaring the bucket name in some `constants.java` file. This way, the bucket name is known across stacks without introducing stack dependencies.

Reply:
Is having this stack dependency an issue? Since the the StateMachineStack depends on the ECS stacks which depend on the VPC stack anyway, wouldn't the dependency essentially still be there?

Reply:
I see, you're right. In general, minimising stack dependencies is a good idea. This is because there are times where we have to delete and re-deploy stacks. If there are dependencies then we fall in a rabbit hole of deleting a chain of dependencies. 

While this stack still depends on vpc stack through ecsstack, I would still remove the explicit vpc stack dependency (using a constant s3 name). This can let us potentially separate these stacks in the future.